File: past202004/past202004_e.cs
// https://atcoder.jp/contests/past202004-open/submissions/13541330

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());
        var A = Console.ReadLine().Split().Select(int.Parse).ToArray();

        for (int i = 1; i <= N; i++)
        {
            var c = 0;
            var n = A[i - 1];
            while (true)
            {
                c++;
                if (i == n)
                {
                    break;
                }
                n = A[n - 1];
            }
            Console.Write($"{c} ");
        }
    }
}

File: past202004/past202004_e.fs
// https://atcoder.jp/contests/past202004-open/submissions/13540536

let N = stdin.ReadLine() |> int
let A = stdin.ReadLine().Split() |> Array.map int

let solver i =
    let rec loop c n =
        if i = n then c
        else loop (c + 1) A.[n - 1]
    loop 1 A.[i - 1]

[ 1..N ]
|> List.map (fun i -> solver i)
|> List.iter (printf "%d ")
